src/schemes/fluid/entropically_damped_sph/system.jl

```diff
@@ -58,6 +60,8 @@ struct EntropicallyDampedSPHSystem{NDIMS, ELTYPE <: Real, IC, M, DC, K, V, TV,
     pressure_acceleration_formulation :: PF
     transport_velocity                :: TV
     source_terms                      :: ST
+    surface_tension                   :: SRFT
+    surface_normal_method             :: SRFN
```

Comment:
Wouldn't it make sense to store the normal method inside the surface tension to avoid cluttering the struct?

Reply:
While the surface normals are only used for surface tension, at this time, they are never actually called from the actual surface tension model. The surface normal calculation steps are always called from the systems.

Reply:
The surface normal calculations can be used in other things than the surface tension. 
1. Tracking of the interface 
2. Heatflux between phases
3. Free surface boundary conditions

Moving it to surface tension just makes it seem like it somehow depends on anything in surface tension but it actually doesn't. Also you need to actually declare surface normal methods for the solids as well but they don't have a surface tension. Declaring normal methods for solids is part of #666 since they are needed for the wetting model. These are than also different for elastic solids and rigid solids.

Reply:
For example for a solid flat plate I declare:
boundary_system = BoundarySPHSystem(tank.boundary, boundary_model,
                                    adhesion_coefficient=1.0, surface_normal_method=StaticNormals((0.0, 1.0)))

Reply:
So maybe we should rename this `normal_method` instead since this works better when applied to solid/boundaries?

Reply:
Currently there will be:
For fluids:
ColorfieldSurfaceNormal
And for Bnd:
StaticNormal (constant value for flat surfaces)
PrecalculatedNormal (curved surfaces)
ColorfieldSurfaceNormal (TLSPH)

Reply:
Then I agree it makes sense that each system has a `surface_normal_method`.